### PR TITLE
Allows grenades to chain explode

### PR DESCRIPTION
**Pull request by SmArtKar**
## About The Pull Request

Allows grenades to chain explode. That's it. When something goes boom your grenade goes boom. Do I really need to explain this?

## Why It's Good For The Game

More funny gimmicks and tricks

## Changelog
:cl:
balance: Grenades now can chain explode
/:cl:

### CODE REVIEW

**Comment by rdragan** [comment]
Are you sure you've thought deeply of the consequences of this?

**Comment by SmArtKar** [comment]
> Are you sure you've thought deeply of the consequences of this?

Yes.

**Comment by Pugie** [comment]
Heavy explosions already can set off grenades???

**Comment by SmArtKar** [comment]
> Heavy explosions already can set off grenades???

Nope?

**Comment by necromanceranne** [comment]
> > Heavy explosions already can set off grenades???
> 
> Nope?

Yes. It ALREADY does this and has done for a while.

**Comment by moocowswag** [comment]
What about nonvolatile grenades? Do they detonate when hit by an explosion? Like say a smoke bomb or a flash bang.

**Comment by Serxule** [comment]
> What about nonvolatile chem grenades? Do they detonate when hit by an explosion? Like say a smoke bomb or a flash bang.

only heavy explosions set off non-explosive grenades as far as I know and have seen, but normal explosive grenades do already chain react

**Comment by ATH1909** [comment]
> > > Heavy explosions already can set off grenades???
> > 
> > 
> > Nope?
> 
> Yes. It ALREADY does this and has done for a while. People have been using boxes of the same explosive to chain them together and create a more severe explosion than what just one bomb can produce for some time now.

Do you have a link to the code for this?

**Comment by necromanceranne** [comment]
> > > > Heavy explosions already can set off grenades???
> > > 
> > > 
> > > Nope?
> > 
> > 
> > Yes. It ALREADY does this and has done for a while. People have been using boxes of the same explosive to chain them together and create a more severe explosion than what just one bomb can produce for some time now.
> 
> Do you have a link to the code for this?

I really wish I did but I swear on me mum this is a real thing and is observable behaviour with a few grenades. It may not be 100% guaranteed however, and likely has to do with ex_act being called on the contents of objects and mobs, with grenades included.

**Comment by Mickyan** [comment]
Afaik it's only the case for explosives on the same tile as the explosion epicenter because of this horrible hack right here https://github.com/tgstation/tgstation/blob/fb488fdfff3a22b2803fc9a824680651b5b86d4e/code/controllers/subsystem/explosions.dm#L335-L358

**Comment by Mothblocks** [comment]
wow lol

File: code/game/objects/items/grenades/grenade.dm [hunk 1]
```diff
@@ -44,6 +44,9 @@
 	sleep(det_time)//so you dont die instantly
 	return BRUTELOSS
 
+/obj/item/grenade/ex_act(severity, target) // Little boom can chain a big boom.
+	detonate()
```
Comment (Mothblocks) [inline code/game/objects/items/grenades/grenade.dm:48]: Call `..()`

Comment (Mothblocks) [reply code/game/objects/items/grenades/grenade.dm:48]: Or hell, even just register to `COMSIG_ATOM_EX_ACT` with `detonate` yourself if you want after making `detonate` a SIGNAL_HANDLER.

**Comment by 81Denton** [comment]
@SmArtKar This PR has been stale for a while, please ping me or any other maintainer when you have time to address Mothblocks' review.

**Pull request closed without merging**